components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -139,37 +144,73 @@ class Nimbus : NimbusApi {
 
         // Do initialization off of the main thread
         scope.launch {
-            // Get experiments
-            val activeExperiments = nimbus.getActiveExperiments()
-
-            // Record enrollments in telemetry
-            recordExperimentTelemetry(activeExperiments)
-
+            tryGetActiveExperiments()?.let {
+                recordExperimentTelemetry(it)
+                onExperimentUpdated?.invoke(it)
+            }
             isInitialized = true
-
-            // Invoke the callback with the list of active experiments for the consuming app to
-            // process.
-            onExperimentUpdated?.invoke(activeExperiments)
         }
     }
 
     override fun getActiveExperiments(): List<EnrolledExperiment> =
-        if (isInitialized) { nimbus.getActiveExperiments() } else { emptyList() }
+        if (isInitialized) { tryGetActiveExperiments().orEmpty() } else { emptyList() }
 
     override fun getExperimentBranch(experimentId: String): String? =
         if (isInitialized) { nimbus.getExperimentBranch(experimentId) } else { null }
 
     override fun updateExperiments() {
         if (!isInitialized) return
-        nimbus.updateExperiments()
-        onExperimentUpdated?.invoke(nimbus.getActiveExperiments())
+        // We need to launch this off of the main thread because it involves a network request
+        // within the nimbus-sdk.
+        scope.launch {
+            if (tryUpdateExperiments()) {
+                tryGetActiveExperiments()?.let {
+                    recordExperimentTelemetry(it)
+                    onExperimentUpdated?.invoke(it)
+                }
```

Comment:
Nit: duplicated code.

Suggest putting this in a nascent `onExperimentsUpdated()` method.